Apps/W1/HybridGP/app/src/pages/GPMigrationConfiguration.Page.al

```diff
@@ -363,6 +340,124 @@ page 4050 "GP Migration Configuration"
                 }
             }
 
+            group(HistoricalData)
+            {
+                Caption = 'Historical Snapshot';
+                InstructionalText = 'Choose whether to migrate detailed transactions from GP. These transactions will be placed in separate historical tables and visible in specific GP list pages.';
+
+                group(HistoricalMain)
+                {
+                    ShowCaption = false;
+
+                    field("EnableDisable Historical Trx."; EnableDisableAllHistTrx)
+                    {
+                        Caption = 'Enable/Disable All Transactions';
+                        ToolTip = 'Specifies whether to migrate historical transactions.';
+
+                        trigger OnValidate()
+                        begin
+                            PrepSettingsForFieldUpdate();
+
+                            Rec.Validate("Migrate Hist. GL Trx.", EnableDisableAllHistTrx);
+                            Rec.Validate("Migrate Hist. AR Trx.", EnableDisableAllHistTrx);
+                            Rec.Validate("Migrate Hist. AP Trx.", EnableDisableAllHistTrx);
+                            Rec.Validate("Migrate Hist. Inv. Trx.", EnableDisableAllHistTrx);
+                            Rec.Validate("Migrate Hist. Purch. Trx.", EnableDisableAllHistTrx);
+
+                            repeat
+                                GPCompanyAdditionalSettings.Validate("Migrate Hist. GL Trx.", EnableDisableAllHistTrx);
+                                GPCompanyAdditionalSettings.Validate("Migrate Hist. AR Trx.", EnableDisableAllHistTrx);
+                                GPCompanyAdditionalSettings.Validate("Migrate Hist. AP Trx.", EnableDisableAllHistTrx);
+                                GPCompanyAdditionalSettings.Validate("Migrate Hist. Inv. Trx.", EnableDisableAllHistTrx);
+                                GPCompanyAdditionalSettings.Validate("Migrate Hist. Purch. Trx.", EnableDisableAllHistTrx);
+                                GPCompanyAdditionalSettings.Modify();
+                            until GPCompanyAdditionalSettings.Next() = 0;
+                        end;
+                    }
+                }
+
+                group(HistoricalAreas)
+                {
+                    ShowCaption = false;
+
+                    field("Migrate Hist. GL Trx."; Rec."Migrate Hist. GL Trx.")
+                    {
+                        Caption = 'GL Transactions';
+                        ToolTip = 'Specifies whether to migrate Historical GL transactions.';
+
+                        trigger OnValidate()
+                        begin
+                            PrepSettingsForFieldUpdate();
+
+                            repeat
```

Comment:
There should be if GPCompanyAdditionalSettings.FindSet() then before the loop

Reply:
Same for all below

Reply:
I am, it's done in PrepSettingsForFieldUpdate().

Reply:
Ideally the PrepSettingsForFieldUpdate should return a boolean to indicate if there are records or not. 
I assume we currently trust that the records will be there. No need to fix it with this PR, it is a more general comment.